### PR TITLE
[BUG] cargo make test fails out of the box

### PR DESCRIPTION
**Pull request by reta**
Signed-off-by: Andriy Redko <andriy.redko@aiven.io>

### Description
- there cert tests were failing because client and server used different CA (fixed)
    ```
    running 8 tests
    test full_certificate_ca_chain_validation ... ok
    test full_certificate_ca_validation ... ok
    test certificate_certificate_ca_validation ... ok
    test none_certificate_validation ... ok
    test default_certificate_validation ... ok
    test fail_certificate_certificate_validation ... ok
    test full_certificate_validation ... ok
    test certificate_certificate_validation ... ok
    
    test result: ok. 8 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 1.09s
    ```
- the Dockerfile is not necessary because the `security` plugin could be disabled using `DISABLE_SECURITY_PLUGIN` (fixed)

### Issues Resolved
Closes https://github.com/opensearch-project/opensearch-rs/issues/45

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .ci/certs/testnode-key.pem [hunk 1]
```diff
@@ -0,0 +1,118 @@
+Private-Key: (2048 bit, 2 primes)
```
Comment (reta) [inline .ci/certs/testnode-key.pem:1]: Converting to PEM so the `security` plugin could read it

File: .ci/certs/testnode.pem [hunk 2]
```diff
@@ -0,0 +1,21 @@
+-----BEGIN CERTIFICATE-----
```
Comment (reta) [inline .ci/certs/testnode.pem:1]: Converting to PEM so the `security` plugin could read it

File: .ci/run-opensearch.sh [hunk 3]
```diff
@@ -47,6 +59,9 @@ END
   volume_name=${node_name}-rest-test-data
   volumes+=($(cat <<-END
     --volume $volume_name:/usr/share/opensearch/data${i}
+    --volume $ssl_ca_pem:/usr/share/opensearch/config/root-ca.pem
```
Comment (reta) [inline .ci/run-opensearch.sh:62]: Using the same CA as the client (plus node certs for server)

File: .ci/run-opensearch.sh [hunk 4]
```diff
@@ -80,7 +87,7 @@ END
   docker run \
     --name "$node_name" \
     --network "$network_name" \
-    --env "ES_JAVA_OPTS=-Xms1g -Xmx1g" \
+    --env "OPENSEARCH_JAVA_OPTS=-Xms1g -Xmx1g" \
```
Comment (reta) [inline .ci/run-opensearch.sh:90]: Using proper env variable

**Comment by dblock** [comment]
Awesome! Let's run this as part of CI? #112

**Comment by reta** [comment]
Minor semantic mismatch: the test use `SECURE_INTEGRATION=true|false` but plugin setting is reversed `DISABLE_SECURITY_PLUGIN = not SECURE_INTEGRATION` :D

**Comment by dblock** [comment]
> Minor semantic mismatch: the test use `SECURE_INTEGRATION=true|false` but plugin setting is reversed `DISABLE_SECURITY_PLUGIN = not SECURE_INTEGRATION` :D

Things like `DISABLE_SECURITY_PLUGIN` are called negative-booleans ;)

**Comment by reta** [comment]
> > Minor semantic mismatch: the test use `SECURE_INTEGRATION=true|false` but plugin setting is reversed `DISABLE_SECURITY_PLUGIN = not SECURE_INTEGRATION` :D
> 
> Things like `DISABLE_SECURITY_PLUGIN` are called negative-booleans ;)

Haha yeah 👍

**Comment by reta** [comment]
> The issue in #45 was somewhat different and was fixed by excluding tests that required a host. We also have #35 and some other similar/related issues - let's see which ones this fixes.

Apologies @dblock , the GA/CI seems to be not liking the certs as volumes, reverting this part to reusing the custom docker file :(

**Comment by dblock** [comment]
So without this change I can run `cargo make test` without problems because we added the `_integration` feature and that's not on by default.

I believe this will fix `cargo test --features _integration`. We should also then re-add it to `cargo make test` as part of this PR.

**Comment by reta** [comment]
> cargo test --features _integration

Correct, we should not need `--features _integration` after that

**Comment by dblock** [comment]
> > cargo test --features _integration
> 
> Correct, we should not need `--features _integration` after that

We may still need it if we want `cargo test` without arguments to still work (default to unit tests, no docker).

**Comment by dblock** [comment]
Thanks for fixing these. I merged, but maybe we/you want to add a CI that runs this?

**Comment by reta** [comment]
> We may still need it if we want `cargo test` without arguments to still work (default to unit tests, no docker).

Picking this one up right away, I know how to make it work

**Comment by reta** [comment]
> Thanks for fixing these. I merged, but maybe we/you want to add a CI that runs this?

Thanks @dblock , yes, on it :)

**Pull request merged**